backend/queue/queue.go

```diff
@@ -277,8 +336,7 @@ func (q *Queue) nackExpiredItems(ctx context.Context, timeout time.Duration) err
 		// If the item has timed out or the client has disconnected, the item is
 		// considered expired and should be moved back to the work queue.
 		if time.Since(itemTimestamp) > timeout || ctx.Err() != nil {
-
-			err = q.swapLane(ctx, string(item.Key), item.ModRevision, string(item.Value), q.work)
+			err = q.swapLane(ctx, string(item.Key), item.ModRevision, string(item.Value), q.workPrefix())
```

Comment:
This moves queue items from the "in-flight" state to the "ready" state. It's intended to handle cases where an item was de-queued, but never successfully Acked. After 60 seconds (hard-coded timeout for now), items will be moved from "in-flight" to "ready".

Use case: there was an error processing the queue item, and it needs to be retried.

I'm not sure what you mean by "expired check request", because I don't think adhoc checks actually have an expiry.

Thinking about this again, I'm not so sure we want automatic nack. Perhaps this should be removed.

Reply:
Since we only clean-up on dequeue, there is the possibility that during a partition an older ad hoc check request will pile up in a minority backend's queue. What happens when the partition is healed?

Reply:
When the partition is healed, outstanding garbage will be cleaned up the next time an item is enqueued.

Reply:
Won't the minority backend, upon reconnecting, execute the (now stale) request, potentially overwriting a more recent event?

Reply:
There could not be a more recent event, because that would imply another enqueue has occurred during the meantime, in which case the garbage would get cleaned up.

It could be that a scheduled check would get executed after the adhoc check is issued, and then the adhoc check would overwrite the result of that, but that could happen in any event, even without a network partition.

Reply:
There can be a more recent event, because the client could have reconnected to a majority member backend and have executed either another ad hoc check request or a scheduled check request leading to a newer event.

Reply:
Oh. Also, my example is invalid, because the client wouldn't be connected to the backend with the stale event.

yeah, problems here take some serious contrivances. we're good.